pandas/tests/series/methods/test_astype.py

```diff
@@ -107,6 +107,20 @@ def test_astype_dict_like(self, dtype_class):
 
 
 class TestAstype:
+    def test_astype_object_to_dt64_non_nano(self):
+        # GH#55756
+        ts = Timestamp("2999-01-01")
+        dtype = "M8[us]"
+        # NB: the 2500 is interpreted as nanoseconds and rounded *down*
+        # to 2 microseconds
+        vals = [ts, "2999-01-02 03:04:05.678910", 2500]
```

Comment:
Could we add a doc example/explanation for this edge case (maybe in to_datetime)?

Reply:
to_datetime isn't a great fit bc it doesnt have a dtype keyword. ill have to give some thought to where this might live.  would you be OK with punting on that for the moment?  this is a blocker for resolution-inference

Reply:
OK to punt for now. I do think this is a potential non-obvious behavior that users might encounter so would be good to document in the future